CMakeLists.txt

```diff
@@ -167,126 +168,155 @@ IF(BUILD_STATIC_LIB)
 ELSE()
   ADD_LIBRARY(binaryen SHARED ${binaryen_SOURCES})
 ENDIF()
-TARGET_LINK_LIBRARIES(binaryen passes wasm asmjs emscripten-optimizer ir cfg support)
-INSTALL(TARGETS binaryen DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
-INSTALL(FILES src/binaryen-c.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+set(_dependencies "passes;wasm;asmjs;emscripten-optimizer;ir;cfg;support")
+foreach(_dep ${_dependencies})
+  SET_TARGET_PROPERTIES(${_dep} PROPERTIES OUTPUT_NAME binaryen-${_dep})
+  TARGET_INCLUDE_DIRECTORIES(${_dep} INTERFACE
+    $<INSTALL_INTERFACE:include/binaryen>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+    )
+
+endforeach()
+
+TARGET_LINK_LIBRARIES(binaryen ${_dependencies})
+install(TARGETS binaryen
+  EXPORT binaryenConfig
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  FRAMEWORK DESTINATION ${CMAKE_INSTALL_PREFIX}
```

Comment:
Seems like this whole block is duplicated in a lot of places.. and that these values should really be the defaults.  Are they not?